From 0ce9a8232b1c476edd388b3ed1187f229a0dc484 Mon Sep 17 00:00:00 2001
From: Byungchul Park <byungchul.park@lge.com>
Date: Mon, 9 Jan 2023 12:33:48 +0900
Subject: [PATCH 20/23] dept: Apply timeout consideration to waitqueue wait

Now that CONFIG_DEPT_AGGRESSIVE_TIMEOUT_WAIT was introduced, apply the
consideration to waitqueue wait, assuming an input 'ret' in
___wait_event() macro is used as a timeout value.

Signed-off-by: Byungchul Park <byungchul.park@lge.com>
---
 include/linux/wait.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/linux/wait.h b/include/linux/wait.h
index ede466c27..87888ee52 100644
--- a/include/linux/wait.h
+++ b/include/linux/wait.h
@@ -304,7 +304,10 @@ extern void init_wait_entry(struct wait_queue_entry *wq_entry, int flags);
 	struct wait_queue_entry __wq_entry;					\
 	long __ret = ret;	/* explicit shadow */				\
 										\
-	sdt_might_sleep_weak(NULL);						\
+	if (!__ret || __ret == MAX_SCHEDULE_TIMEOUT)				\
+		sdt_might_sleep_weak(NULL);					\
+	else									\
+		sdt_might_sleep_weak_timeout(NULL);				\
 	init_wait_entry(&__wq_entry, exclusive ? WQ_FLAG_EXCLUSIVE : 0);	\
 	for (;;) {								\
 		long __int = prepare_to_wait_event(&wq_head, &__wq_entry, state);\
-- 
2.39.0.rc2.1.gbd5df96b79

